spec/fake_consumer_interceptor.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require "kafka/consumer_interceptor"
+
+class FakeConsumerInterceptor < Kafka::ConsumerInterceptor
+  def initialize(append_s = '')
+    @init_count = 1
+    @close_count = 0
+    @append_s = append_s
+  end
+
+  def on_consume(batches:)
+    intercepted_batches = []
+    batches.each do |batch|
+      intercepted_messages = []
+      batch.messages.each do |message|
+        intercepted_messages.append(Kafka::FetchedMessage.new(
+          message: Kafka::Protocol::Message.new(
+            value: message.value + @append_s,
+            key: message.key,
+            create_time: message.create_time,
+            offset: message.offset
+          ),
+          topic: message.topic,
+          partition: message.partition
+        ))
+      end
+      intercepted_batches.append(Kafka::FetchedBatch.new(
+        topic: batch.topic,
+        partition: batch.partition,
+        highwater_mark_offset: batch.highwater_mark_offset,
+        messages: intercepted_messages,
+        last_offset: batch.last_offset,
+        leader_epoch: batch.leader_epoch
+      ))
+    end
+
+    intercepted_batches
+  end
+
+  def close
+    @close_count = @close_count + 1
```

Comment:
Ruby allows `@close_count += 1`, unless there's a style-guide reason to not use it